File: bms/go/geo/geo_r2_Rect_Contains/geo_r2_Rect_Contains.go
package test

var ()

const ()

func (r geo_r2_Rect) Contains(other geo_r2_Rect) bool {
	return r.X.ContainsInterval(other.X) && r.Y.ContainsInterval(other.Y)
}

func (i geo_r1_Interval) ContainsInterval(oi geo_r1_Interval) bool {
	if oi.IsEmpty() {
		return true
	}
	return i.Lo <= oi.Lo && oi.Hi <= i.Hi
}

func (i geo_r1_Interval) IsEmpty() bool { return i.Lo > i.Hi }

type geo_r2_Rect struct {
	X, Y geo_r1_Interval
}

type geo_r1_Interval struct {
	Lo, Hi float64
}

File: translations_only/success/go/geo/geo_r2_Rect_Contains/gpt4/Hinted/geo_r2_Rect_Contains.rs

struct GeoR2Rect {
    x: GeoR1Interval,
    y: GeoR1Interval,
}

struct GeoR1Interval {
    lo: f64,
    hi: f64,
}

impl GeoR1Interval {
    fn new(lo: f64, hi: f64) -> Self {
        GeoR1Interval { lo, hi }
    }

    fn is_empty(&self) -> bool {
        self.lo > self.hi
    }
}

fn contains_interval(i: &GeoR1Interval, oi: &GeoR1Interval) -> bool {
    if oi.is_empty() {
        return true;
    }
    i.lo <= oi.lo && oi.hi <= i.hi
}

fn contains(r: &GeoR2Rect, other: &GeoR2Rect) -> bool {
    contains_interval(&r.x, &other.x) && contains_interval(&r.y, &other.y)
}

fn main() {}
